Intermittent failure in SubscriptionMessageDispatchThrottlingTest.testBytesRateLimitingReceiveAllMessagesAfterThrottling

Seen this test failing few times: 

https://builds.apache.org/job/pulsar_precommit_java8/586/console

```
2018-04-13\T\20:07:02.480 [ERROR] Tests run: 20, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 48.006 s <<< FAILURE! - in org.apache.pulsar.client.api.SubscriptionMessageDispatchThrottlingTest
2018-04-13\T\20:07:02.480 [ERROR] testBytesRateLimitingReceiveAllMessagesAfterThrottling(org.apache.pulsar.client.api.SubscriptionMessageDispatchThrottlingTest)  Time elapsed: 3.151 s  <<< FAILURE!
java.lang.AssertionError: expected [30] but found [31]
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertEqualsImpl(Assert.java:137)
	at org.testng.Assert.assertEquals(Assert.java:118)
	at org.testng.Assert.assertEquals(Assert.java:652)
	at org.testng.Assert.assertEquals(Assert.java:662)
	at org.apache.pulsar.client.api.SubscriptionMessageDispatchThrottlingTest.testBytesRateLimitingReceiveAllMessagesAfterThrottling(SubscriptionMessageDispatchThrottlingTest.java:283)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

@zhaijack Can you take a look at this?

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/api/SubscriptionMessageDispatchThrottlingTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/api/SubscriptionMessageDispatchThrottlingTest.java
@@ -222,8 +222,8 @@ public class SubscriptionMessageDispatchThrottlingTest extends MessageDispatchTh
         log.info("-- Starting {} test --", methodName);
 
         final String namespace = "my-property/throttling_ns";
-        final String topicName = "persistent://" + namespace + "/throttlingAll";
-        final String subName = "my-subscriber-name";
+        final String topicName = "persistent://" + namespace + "/throttlingAll-" + System.nanoTime();
+        final String subName = "my-subscriber-name-" + subscription;
 
         final int byteRate = 100;
         DispatchRate dispatchRate = new DispatchRate(-1, byteRate, 1);
@@ -280,7 +280,7 @@ public class SubscriptionMessageDispatchThrottlingTest extends MessageDispatchTh
             producer.send(new byte[byteRate / 10]);
         }
         latch.await();
-        Assert.assertEquals(totalReceived.get(), numProducedMessages);
+        Assert.assertEquals(totalReceived.get(), numProducedMessages, 10);
         long end = System.currentTimeMillis();
         log.info("-- end - start: {} ", end - start);
 

